charms/knative-serving/config.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+options:
+  namespace:
+    default: ""
+    description: The namespace to deploy Serving to.  If left blank, this will be the same as this charm’s model name
+    type: string
+  domain.name:
+    default: "10.64.140.43.nip.io"
```

Comment:
Should we leave this address as the default one? Is there a programmatic way of getting this?

Reply:
Also I'm not sure if we should be using `.` for config naming.

Reply:
hmm maybe there is a programmatic way, but I don't know it.  It depends on how the load balancer is set up and the external IP that it is going to be given right?

We could also make this required or blank, but they also feel bad:
* required is kind of wrong: this value isn't actually required for knative to operate (you can still access services from inside the cluster, and possibly from outside too but I'm not sure)
* blank could work if we handled blank as a `None` and passed null to the manifests.  

But I think the least-crappy default is `"10.64.140.43.nip.io"` which at least aligns with our default install method (microk8s + metallb)

Reply:
Re the `.` in config naming, it should be syntactically correct but I understand it feels ugly.  What alternative are you thinking?  

In general, we're mapping a subset of a dictionary of settings (the knative-operator CRDs) to a flat naming convention and (since it will cause havoc in upgrades) we want to avoid ever changing a config name, so to me that means the easiest/safest approach is just use knative's nomenclature directly.  But it is ugly